test/integration/index_management.test.js

```diff
@@ -258,39 +271,106 @@ describe('Indexes', function () {
 
-                collection.indexInformation(function (err, result) {
-                  test.ok(result['a_1'] == null);
-                  client.close(done);
-                });
-              });
-            }
-          );
+    afterEach(async function () {
+      await db.dropCollection('test_drop_indexes');
+    });
+
+    it('should return true and should no longer exist in the collection', async function () {
+      // Drop all the indexes
+      const result = await collection.dropIndexes();
+      expect(result).to.equal(true);
+
+      const res = await collection.indexInformation();
+      expect(res['a_1']).to.equal(undefined);
+    });
+  });
+
+  context('when dropIndexes fails', function () {
+    let collection;
+
+    beforeEach(async function () {
+      collection = await db.createCollection('test_drop_indexes');
+      await collection.insert({ a: 1 });
+      // Create an index on the collection
+      await db.createIndex(collection.collectionName, 'a');
+      /**@type {import('../tools/utils').FailPoint} */
+      await client
+        .db()
+        .admin()
+        .command({
+          configureFailPoint: 'failCommand',
+          mode: {
+            times: 1
+          },
+          data: {
+            failCommands: ['dropIndexes'],
+            errorCode: 91
+          }
         });
-      });
-    }
+    });
+
+    afterEach(async function () {
+      await db.dropCollection('test_drop_indexes');
+    });
+
+    it('should return false', {
+      metadata: {
+        requires: {
+          mongodb: '>4.0'
+        }
+      },
+
+      test: async function () {
+        const result = await collection.dropIndexes();
+        expect(result).to.equal(false);
+      }
+    });
+  });
+
+  context('indexExists', function () {
+    let collection;
+
+    beforeEach(async function () {
+      collection = await db.createCollection('test_index_exists');
+      await collection.insert({ a: 1 });
+
+      await db.createIndex(collection.collectionName, 'a');
+      await db.createIndex(collection.collectionName, ['c', 'd', 'e']);
+    });
+
+    afterEach(async function () {
+      await db.dropCollection('test_index_exists');
+    });
+
+    it('should return true when index of type string exists', async function () {
+      const result = await collection.indexExists('a_1');
+      expect(result).to.equal(true);
+    });
+
+    it('should return false when index of type string does not exist', async function () {
+      const result = await collection.indexExists('b_2');
+      expect(result).to.equal(false);
+    });
+
+    it('should return true when an array of indexes exists', async function () {
+      const result = await collection.indexExists('c_1_d_1_e_1');
+      expect(result).to.equal(true);
+    });
+
+    it('should return false when an array of indexes does not exist', async function () {
+      const result = await collection.indexExists('d_1_e_1_f_1');
+      expect(result).to.equal(false);
+    });
```

Comment:
`indexExists` can take an array of indexes (i.e., `indexExists(['a_1', 'b_2'])`) and returns true iff all the indexes in the array exist.  That is the scenario we want to test with these two tests

Reply:
When you create the index with an array, though, the index looks like `'c_1_d_1_e_1'`. So when you check it like `indexExists(['a_1', 'b_2'])`, wouldn't it always return false?